src/commands/check/index.ts

```diff
@@ -98,7 +121,19 @@ export default class Check extends SwankyCommand<typeof Check> {
       {
         title: "Verify ink version",
         task: async (ctx) => {
-          const supportedInk = ctx.swankyConfig?.node.supportedInk;
+          let supportedInk = ctx.swankyConfig?.node.supportedInk;
+          const regex = /cargo-contract-contract (.*)-unknown-x86_64-unknown-linux-gnu/;
+          const cargoContract = ctx.versions.tools.cargoContract;
+          if (cargoContract) {
+            const match = cargoContract.match(regex);
+            if (match) {
+              const cargoVersion = match[1];
+              const version = cargoContractDeps.get(cargoVersion);
+              if (version && semver.gt(supportedInk!, version[0])) {
```

Comment:
Checking order is wrong here, version goes first. Also the higher version should be the one supported:

`semver.gt(versions[versions.length - 1], supportedInk!)`